docs/multi-stage-query/reference.md

```diff
@@ -556,6 +556,31 @@ The following ISO 8601 periods are supported for `TIME_FLOOR` and the string con
 - P3M
 - P1Y
 
+The string constant can also include any of the keywords mentioned above:
+
+- `HOUR` - Same as `'PT1H'`
+- `DAY` - Same as `'P1D'`
+- `MONTH` - Same as `'P1M'`
+- `YEAR` - Same as `'P1Y'`
+- `ALL TIME`
+- `ALL` - Alias for `ALL TIME`
+
+Examples:
+
+```SQL
+-- Keyword
+PARTITIONED BY HOUR
+
+-- String constant
+PARTITIONED BY 'HOUR'
+
+-- Or
+PARTITIOND BY 'PT1H'
```

Comment:
Would it possible to make this not work? Like if the user wants to do a `PARTITIOND BY <period>` they have to do `PARTITIONED BY TIME_FLOOR(__time, <period>)` the reasoning for this is that people should really be sticking to the existing partition types. When people partition by whacky periods they are asking for trouble...

One of the biggest offenders of this that I have encountered is users wanting to partition by "week" (P1W or P7D). It seems like a cool zany middle ground between day and month, but in reality it is asking for trouble later on.

Reply:
Great comment. As it turns out, allowing a string doesn't change much. The user can still say `TIME_FLOOR(__time, 'P1W')`. There is code within the new validation code to accept a limited set of times. So, if we don't like `P1W`, we simply don't accept it. The code currently accepts only the strings that match the SQL symbols we already support.

This change, by the way is a "nice to have" to support moving validation out of ad-hoc code into the Calcite validator. Basically, Calcite literals know about strings, but not Druid `Granularities` so we now translate the SQL literals to strings, and convert those literals to a `Granularity` during validation. Previously, we created a `Granularity` in the parser as a "special" field, but that doesn't play by the Calcite rules. Easy enough to not accept a string literal in the parser; leave it as the SQL keywords and `TIME_FLOOR`.

Note that "partitioning" is defined as time partitioning. Thus, we don't really need the `TIME_FLOOR(__time` part of the expression: we know that partitioning means `__time` and and means using `TIME_FLOOR`.

Also, in the catalog, partitioning is defined as an ISO 8601 string: `P1D`. Given that, allowing the same form in the SQL syntax isn't quite as crazy as it might first look. The catalog, also, validates the string to ensure it is one of the supported forms.

That said, aside from the code, do we have a list of the "actually" supported forms? Does anyone actually partition by second or minute, say?